docs/decisions/0006-versioning-of-event-transformers.rst

```diff
@@ -0,0 +1,34 @@
+Versioning of event transformers
+================================
+
+Status
+------
+
+Pending
+
+Context
+-------
+
+Event transformers may undergo modification in future in response to consumer request, change in specification, bug fixes etc.
+
+Decision
+--------
+#. Versions of event transformers will be maintained and emitted as part of  xAPI and Caliper events.
+
+#. Transformer version will be defined by two non-negative integers X and Y representing major and minor version respectively in the form: X.Y.
+
+#. Major version (X) will be incremented when:
+
+   #. Transformer is changed due to update in original specification (xAPI or Caliper).
+
+   #. A key is removed from or renamed in the existing transformer.
+
+   #. Value of a key is updated in the existing transformer.
+
+#. Minor version (Y) will be incremented when:
+
+   #. A key is added to an existing transformer.
+
+#. Change logs of transformers will be maintained for both xAPI and Caliper.
+
+#. This version (X.Y) can be found in `` context [ extensions [ minorVersion ] ]`` for xAPI statement and in ``extensions [ minorVersion ]`` for Caliper event.
```

Comment:
[naming confusion] Since X, the major version, will also be included in this `extensions` field, can we rename the field to be simply `version` instead of `minorVersion`?

Reply:
The major version here does not refer to specification version because:
1. xAPI spec requires a "version" field in the statement, that needs to be set by LRS. But the spec also states that "Learning Record Providers SHOULD NOT set the Statement version."
2. Caliper specs requires learning record providers to include a Caliper specification version in the event in this format: _"dataVersion": "http://purl.imsglobal.org/ctx/caliper/v1p1"_.

Reply:
Yes, I understand that the spec version is different from this Open edX specific version we are using.

My understanding from what you've written here is that we will be stamping both X and Y in this new extensions field. Is that correct?  If so, the name of the extensions field should be generic so it can contain both X and Y. Calling it `minorVersion` implies that it will contain only the Y value.

Reply:
I see what you mean. How about if we name it to `transformerVersion`?

Reply:
`transformer` is an implementation detail that we don't need to expose to event recipients.
How about `eventVersion`?